src/anemoi/inference/condition.py

```diff
@@ -0,0 +1,285 @@
+
+        Examples
+        --------
+        >>> ds = xr.tutorial.load_dataset("air_temperature")
+        >>> ds
+        <xarray.Dataset> Size: 31MB
+        Dimensions:  (lat: 25, time: 2920, lon: 53)
+        Coordinates:
+        * lat      (lat) float32 100B 75.0 72.5 70.0 67.5 65.0 ... 22.5 20.0 17.5 15.0
+        * lon      (lon) float32 212B 200.0 202.5 205.0 207.5 ... 325.0 327.5 330.0
+        * time     (time) datetime64[ns] 23kB 2013-01-01 ... 2014-12-31T18:00:00
+        Data variables:
+            air      (time, lat, lon) float64 31MB ...
+        Attributes:
+            Conventions:  COARDS
+            title:        4x daily NMC reanalysis (1948)
+            description:  Data is from NMC initialized reanalysis\n(4x/day).  These a...
+            platform:     Model
+            references:   http://www.esrl.noaa.gov/psd/data/gridded/data.ncep.reanaly...
+        >>> Condition.from_xarray(ds))
+        Condition(['air'], private_info = None)
+        >>> Condition.from_xarray(ds, flatten="{variable}_{lat}"))
+        Condition(['air_75.0', 'air_72.5', 'air_70.0', 'air_67.5', 'air_65.0', 'air_62.5', 'air_60.0', 'air_57.5', 'air_55.0', 'air_52.5', 'air_50.0', 'air_47.5', 'air_45.0', 'air_42.5', 'air_40.0', 'air_37.5', 'air_35.0', 'air_32.5', 'air_30.0', 'air_27.5', 'air_25.0', 'air_22.5', 'air_20.0', 'air_17.5', 'air_15.0'], private_info = None)
+        """
+
+        # Flatten the data if required
+        if flatten is not None:
+            keys = extract_keys(flatten)
+            if len(keys) == 0:
+                raise ValueError("No keys found in flatten")
+
+            if variable_dim not in keys:
+                raise KeyError(f"variable_dim {variable_dim} not found in keys {keys}")
+
+            new_data = xr.Dataset()
+            for var in data.coords[variable_dim].values:
+                var_ds = data.sel({variable_dim: var})
+
+                if not any(key in var_ds.dims for key in keys):
+                    new_data[var] = data.sel({variable_dim: var})
+
+                for perm in permute_dict({key: list(np.atleast_1d(var_ds.coords[key].values)) for key in keys}):
+                    new_data[flatten.format(**perm)] = data.sel({variable_dim: var, **perm})
+
+            data = new_data
+
+        # Convert to DataArray if Dataset so dimensions are in the correct order
+        if isinstance(data, xr.Dataset):
+            data = data.to_dataarray(dim=variable_dim)
+
+        dims = list(data.dims)
+        dims.remove(variable_dim)
+        data = data.transpose(variable_dim, *dims)
+
+        # Get all variables and their values in the data
+        variable_dict = {}
+        for var in data.coords[variable_dim].values:
+            variable_dict[var] = data.sel({variable_dim: var}).values
+        return Condition(variable_dict, private_info=private_info)
+
+    @classmethod
+    def from_earthkit(self, fieldlist: "ekd.FieldList", private_info: Any = None, **kwargs) -> Self:
```

Comment:
I like `-> Self` but it is a python >= 3.11 feature, and our minimum supported version is 3.9

So instead, please do `from __future__ import annotations` and return the class name explicitly.

Reply:
Apologies, addressed in upcoming commit